R/Model-methods.R

```diff
@@ -698,6 +698,31 @@ setMethod(
   }
 )
 
+## OneParExpExpPrior ----
+
+#' @describeIn dose compute the dose level reaching a specific target
+#'   probability of the occurrence of a DLE (`x`).
+#'
+#' @aliases dose-OneParExpExpPrior
+#' @export
+#'
+setMethod(
+  f = "dose",
+  signature = signature(
+    x = "numeric",
+    model = "OneParExpExpPrior",
+    samples = "Samples"
+  ),
+  definition = function(x, model, samples) {
+    assert_subset("theta", names(samples@data))
```

Comment:
Would be good to also assert that `theta` is not zero (this may still theoretically happen as the support for the exp. dist. includes 0).

Reply:
@wwojciech That is a potential problem. Theta is sampled from the mcmc. If we check here, it ends with an error, but the user has no chance to change. I am wondering if jags returns 0 for theta.

Reply:
I think it can return `0`, but the probability of that is equal `0`. Still, I would need add this assertion - and the function will fail, which is ok. @danielinteractive  - what do you think?

Reply:
I agree that it fails if it is 0. Could we probably add some tiny value to make the calculation run and issue a warning?

Reply:
but when `theta` is from real samples, it will never be exactly 0, so is this a real issue?

Reply:
yes, `theta` will be a posterior samples, and then I think it will never be exactly `0`, but since we have this `doseFunction` concept which allows for the end user to supply any value for `thet`, including `0`, and even a negative ones, so I would still add the assertion for `theta` here.

Reply:
ok good the user input could really be 0. then yeah I would agree to add assertion. we won't get exact 0 from MCMC

Reply:
Makes sense to me. I did not consider that the user may use the function to calculate dose or prob by himsellf. Will add the check.